lib/angstrom.ml

```diff
@@ -277,19 +277,19 @@ let rec peek_char_fail =
         peek_char_fail.run input' pos' more' fail succ in
       ensure_suspended 1 input pos more fail succ' }
 
-let satisfy f =
+let satisfy ?(failure_msg="satisfy") f =
   { run = fun input pos more fail succ ->
     if pos < Input.length input then
       let c = Input.unsafe_get_char input pos in
       if f c
       then succ input (pos + 1) more c
-      else fail input pos more [] "satisfy"
+      else Printf.ksprintf (fail input pos more []) "%s: %C" failure_msg c
```

Comment:
Reporting the character that failed the predicate seems like a good idea.

Reply:
I've kept this.